app/main.py

```diff
@@ -1 +1,46 @@
-#  write your code here
+class Animal:
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True,
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        food_points = 0
+        if self.is_hungry:
+            food_points = self.appetite
+            print(f"Eating {food_points} food points...")
+            self.is_hungry = False
+        return food_points
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 3, is_hungry=is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 7, is_hungry)
```

Comment:
It's a good practice to explicitly specify the `is_hungry` parameter when calling `super().__init__` for clarity, similar to how it's done in the `Cat` class.